### ISSUE TITLE
Restructure JPA repository context population with context configuring processors

### ISSUE THREAD

**Issue by GuusLieben** [Dockbox-OSS/Hartshorn#875]
### Related problem

The existing approach to populating JPA contexts in repositories was initially designed as PoC, and therefore isn't optimally scalable. This should be resolved now that we have the core infrastructure to optimize this.

### Requested feature

Use the results of #873 and the improved core stability of 22.5 to restructure and improve the JPA context approach. This is not a total refactor of the module, but should open the way to improving the stability and usability of the JPA module.

**Comment by GuusLieben** [comment]
No longer needed after #876
